src/lib.rs

```diff
@@ -525,16 +525,15 @@ impl TOTP {
         }
 
         let path = url.path().trim_start_matches('/');
+        let path = urlencoding::decode(path)
+            .map_err(|_| TotpUrlError::AccountNameDecoding(path.to_string()))?
+            .to_string();
         if path.contains(':') {
             let parts = path.split_once(':').unwrap();
-            issuer = Some(
-                urlencoding::decode(parts.0.to_owned().as_str())
-                    .map_err(|_| TotpUrlError::IssuerDecoding(parts.0.to_owned()))?
-                    .to_string(),
-            );
-            account_name = parts.1.trim_start_matches(':').to_owned();
+            issuer = Some(parts.0.to_owned());
+            account_name = parts.1.to_owned();
         } else {
-            account_name = path.to_owned();
```

Comment:
Since this is already merges I think rebasing from master should take this out of this PR's changes

Reply:
That's weird! Because this is already rebased.

Let me give that another spin.

Reply:
Rebased & force pushed.